sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/main/scala/com/azure/cosmos/spark/CosmosPartitionPlanner.scala

```diff
@@ -127,6 +127,7 @@ private object CosmosPartitionPlanner extends CosmosLoggingTrait {
               }
             }
           })
+          .take(1)
```

Comment:
Any tests that should be added to avoid a regression in the future?

Reply:
Don't really know how to test it. The number of records isn't returned/exposed - and adding a large dataset in unit test wouldn't make sense. Given that the risk of introducing a regression there I think is low I thought it is ok to leave this.

Reply:
Added a unit test to ensure exactly one page is retrieved for each feed range